heed/src/env.rs

```diff
@@ -333,6 +335,49 @@ impl Drop for EnvInner {
     }
 }
 
+/// An helping function that transforms the LMDB types into Rust types (`MDB_val` into slices)
+/// and vice versa, the Rust types into C types (`Ordering` into an integer).
+extern "C" fn custom_key_cmp_wrapper<C: Comparator>(
+    a: *const ffi::MDB_val,
+    b: *const ffi::MDB_val,
+) -> i32 {
+    let a = unsafe { ffi::from_val(*a) };
+    let b = unsafe { ffi::from_val(*b) };
+    match C::compare(a, b) {
+        Ordering::Less => -1,
+        Ordering::Equal => 0,
+        Ordering::Greater => 1,
+    }
+}
+
+/// This is a special implementation of [`LexicographicComparator`] which means that no customized
+/// compare function is set to the underlying LMDB database instance. Its implementation of
+/// [`Comparator::compare`] will not be actually used.
+pub struct NoopComparator;
+
+impl LexicographicComparator for NoopComparator {
+    fn compare_elem(a: u8, b: u8) -> Ordering {
+        a.cmp(&b)
+    }
+
+    fn advance(bytes: &mut Vec<u8>) {
+        match bytes.last_mut() {
+            Some(&mut 255) | None => bytes.push(0),
+            Some(last) => *last += 1,
+        }
+    }
+
+    fn retreat(bytes: &mut Vec<u8>) {
+        match bytes.last_mut() {
+            Some(&mut 0) => {
+                bytes.pop();
+            }
+            Some(last) => *last -= 1,
+            None => panic!("Vec is empty and must not be"),
+        }
+    }
```

Comment:
I thought we were not using these as the `NoopComparator` is not used but we actually use the `advance` and `retreat` methods when doing the prefix methods. Can we add a comment about this, please?

Reply:
Good call! I added it in the comments for `DefaultComparator`